tubular/edx_api.py

```diff
@@ -48,16 +52,37 @@ def get_api_url(self, path):
         if self.append_slash:
             path += '/'
 
-        return urljoin(f"{self.api_base_url}/", path)
-
-    def _request(self, method, url, **kwargs):
-        response = requests.request(method, url, auth=SuppliedJwtAuth(self._access_token), **kwargs)
+        return urljoin(f'{self.api_base_url}/', path)
 
+    def _request(self, method, url, log_404_as_error=True, **kwargs):
         try:
+            response = requests.request(method, url, auth=SuppliedJwtAuth(self._access_token), **kwargs)
             response.raise_for_status()
         except HTTPError as exc:
-            exc.content = response.content
+            import pdb
+            pdb.set_trace()
+            status_code = exc.response.status_code
+
+            if status_code == 404 and not log_404_as_error:
+                # Immediately raise the error so that a 404 isn't logged as an API error in this case.
+                raise HttpDoesNotExistException(text_type(exc))
+            elif status_code == 504:
+                # Differentiate gateway errors so different backoff can be used.
+                raise EdxGatewayTimeoutError(text_type(exc))
+
+            if hasattr(exc, 'content'):
+                LOG.error("API Error: {} with status code: {}".format(text_type(exc), status_code))
```

Comment:
Please provide an example of the exception when `exception` has attribute `content` and why do we need it in our new implementation?

Reply:
thx, updated